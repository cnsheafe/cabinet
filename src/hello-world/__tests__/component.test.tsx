import * as React from 'react'
import { shallow, configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

import { HelloWorldComponent, mapStateToProps } from '../'
configure({ adapter: new Adapter() })

describe('HelloWorldComponent', () => {
  let mockInc = jest.fn()
  beforeEach(() => {
    mockInc = jest.fn()
  })

  it('should match the snapshot', () => {
    const wrapper = shallow(<HelloWorldComponent counter={0} inc={mockInc} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should call increment when button is clicked', () => {
    const wrapper = shallow(<HelloWorldComponent counter={0} inc={mockInc} />)
    wrapper.find('button').simulate('click', () => {
      return
    })
    expect(mockInc).toHaveBeenCalled()
  })
})
