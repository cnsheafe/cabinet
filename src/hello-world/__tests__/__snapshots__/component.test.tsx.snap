// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorldComponent should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    counter={0}
    inc={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
        >
          Increment
        </button>,
        <span>
          Counter: 
          0
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Increment",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Increment",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Counter: ",
            0,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Counter: ",
          0,
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
          >
            Increment
          </button>,
          <span>
            Counter: 
            0
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Increment",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Increment",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Counter: ",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Counter: ",
            0,
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
